packages/block-editor/src/components/block-list-appender/index.js

```diff
@@ -1,63 +1,68 @@
 /**
  * External dependencies
  */
-import { last } from 'lodash';
+import { last, isFunction } from 'lodash';
 
 /**
  * WordPress dependencies
  */
 import { withSelect } from '@wordpress/data';
 import { getDefaultBlockName } from '@wordpress/blocks';
-import { __ } from '@wordpress/i18n';
-import { IconButton } from '@wordpress/components';
 
 /**
  * Internal dependencies
  */
 import IgnoreNestedEvents from '../ignore-nested-events';
 import DefaultBlockAppender from '../default-block-appender';
-import Inserter from '../inserter';
+import ButtonBlockAppender from '../button-block-appender';
 
 function BlockListAppender( {
 	blockClientIds,
 	rootClientId,
 	canInsertDefaultBlock,
 	isLocked,
+	renderAppender,
 } ) {
 	if ( isLocked ) {
 		return null;
 	}
 
-	if ( canInsertDefaultBlock ) {
+	// If auto-insert Blocks is enabled, default to the standard behaviour
+	// of auto-inserting a Block but only if no renderAppender is provided.
+	if ( ! renderAppender && canInsertDefaultBlock ) {
 		return (
-			<IgnoreNestedEvents childHandledEvents={ [ 'onFocus', 'onClick', 'onKeyDown' ] }>
-				<DefaultBlockAppender
+			<div className="block-list-appender">
+				<IgnoreNestedEvents childHandledEvents={ [ 'onFocus', 'onClick', 'onKeyDown' ] }>
+					<DefaultBlockAppender
+						rootClientId={ rootClientId }
+						lastBlockClientId={ last( blockClientIds ) }
+					/>
+				</IgnoreNestedEvents>
+			</div>
+		);
+	}
+
+	// Render prop - custom appender.
+	if ( isFunction( renderAppender ) ) {
```

Comment:
Can we assume renderAppender is always a function and just have if( renderAppender )?

Reply:
We probably can now...